File: apps/web/src/lib/weather.ts
import { apiGet } from "@/lib/api";
import { API_PATHS } from "@/lib/api-paths";

export type WeatherSummary = {
  date: string;
  temp: number | null;
  condition: string;
  icon: string;
  sunset?: string | null;
};

export type WeatherPayload = {
  today: WeatherSummary;
  tomorrow: WeatherSummary;
  weekend: WeatherSummary;
};

export async function fetchWeather(): Promise<WeatherPayload> {
  return apiGet<WeatherPayload>(API_PATHS.weather.summary);
}

File: apps/api/app/schemas/weather.py
from pydantic import BaseModel


class WeatherSummaryOut(BaseModel):
    date: str
    temp: int | None
    condition: str
    icon: str
    sunset: str | None = None


class WeatherPayloadOut(BaseModel):
    today: WeatherSummaryOut
    tomorrow: WeatherSummaryOut
    weekend: WeatherSummaryOut
